telluric/georaster.py

```diff
@@ -1275,12 +1286,17 @@ def copy_with(self, mutable=None, **kwargs):
         """Get a copy of this GeoRaster with some attributes changed. NOTE: image is shallow-copied!"""
         if mutable is None:
             mutable = isinstance(self, MutableGeoRaster)
-        init_args = {'affine': self.affine, 'crs': self.crs, 'band_names': self.band_names, 'nodata': self.nodata_value}
+        init_args = {'affine': self.affine, 'crs': self.crs, 'band_names': self.band_names, 'nodata': self.nodata_value,
+                     'shape': self.shape}
```

Comment:
I believe that the errors in the CI come from adding shape to the init args. In the case where the `copy_with` gets passed a new image with a different shape, there is a mismatch.

Reply:
pushed a fix.

WDYT about the (very partial) pathlib support - should we remove that or leave it in?
(maybe open a separate issue/PR with more extensive solution)

Reply:
I don't see any problem with keeping it. @lilsnore I'm ready to merge the changes and make a new release. Any objections?